sharkfuser/include/fusilli/support/asm_emitter.h

```diff
@@ -554,6 +554,256 @@ inline std::string ConvFPropNode::emitNodePreAsm() const {
+    {0} = torch.aten.permute {0}_perm, {1} : {2}, !torch.list<int> -> {3}
+  )";
+
+  std::string output =
+      std::format(schema,
+                  DW->getValueNameAsm(),       // {0}
+                  "%" + prefix + "_" + suffix, // {1}
+                  DW->getTensorTypeAsm(/*isValueTensor=*/true,
+                                       /*useLogicalDims=*/true), // {2}
+                  DW->getTensorTypeAsm(/*isValueTensor=*/true,
+                                       /*useLogicalDims=*/false) // {3}
+      );
+
+  return oss.str() + output;
+}
+
+// `torch.aten.convolution_backward` requires an input for the weight even when
+// calculating the gradient of the weight. Create an empty tensor with the same
+// dimensions as the weight tensor.
+inline std::string ConvWGradNode::getPermutedEmptyDWOpsAsm() const {
+  std::ostringstream oss;
+  std::string prefix = "empty_DW";
+  std::string suffix = convWGradAttr.getName();
+  std::shared_ptr<TensorAttr> DW = convWGradAttr.getDW();
+
+  // Emit permute dimensions based on layout.
+  oss << getListOfIntOpsAsm(DW->getDim(), prefix, suffix);
+
+  // Emit the permute op itself.
+  constexpr std::string_view schema = R"(
+    %none_DW_{0} = torch.constant.none
+    %dtype_DW_{0} = torch.constant.int {3}
+    %empty_{0} = torch.aten.empty.memory_format {1}, %dtype_DW_{0}, %none_DW_{0}, %none_DW_{0}, %none_DW_{0}, %none_DW_{0} : !torch.list<int>, !torch.int, !torch.none, !torch.none, !torch.none, !torch.none -> {2}
+  )";
+
+  torch_upstream::ScalarType dataType =
+      DataTypeToTorchType.at(DW->getDataType());
+  std::string output =
+      std::format(schema,
+                  suffix,                      // {0}
+                  "%" + prefix + "_" + suffix, // {1}
+                  DW->getTensorTypeAsm(/*isValueTensor=*/true,
+                                       /*useLogicalDims=*/true), // {2}
+                  std::to_string(static_cast<int>(dataType))     // {3}
+      );
+
+  return oss.str() + output;
+}
+
+inline std::string ConvWGradNode::emitNodePreAsm() const {
+  constexpr std::string_view schema = R"(
+    %bias_{0} = torch.constant.none
+    %transposed_{0} = torch.constant.bool false
+    %output_padding_{0} = torch.prim.ListConstruct  : () -> !torch.list<int>
+    %groups_{0} = torch.constant.int 1
+    {1}
+    {2}
+    {3}
+    {4}
+    {5}
+    {11}
+    %true_{0} = torch.constant.bool true
+    %false_{0} = torch.constant.bool false
+    %output_mask_{0} = torch.prim.ListConstruct %false_{0}, %true_{0}, %false_{0} : (!torch.bool, !torch.bool, !torch.bool) -> !torch.list<bool>
+    %grad_input_{0}, {6}_perm, %grad_bias_{0} = torch.aten.convolution_backward {7}, %empty_{0}, %bias_{0}, %stride_{0}, %padding_{0}, %dilation_{0}, %transposed_{0}, %output_padding_{0}, %groups_{0}, %output_mask_{0} : {8}, {12}, !torch.none, !torch.list<int>, !torch.list<int>, !torch.list<int>, !torch.bool, !torch.list<int>, !torch.int, !torch.list<bool> -> !torch.none, {9}, !torch.none
```

Comment:
Just for my understanding - what are the 3 outputs for? Do we only end up using one?